### ISSUE TITLE
False-positive of unused-private-member with class reference

### ISSUE THREAD

**Issue by yumasheta** [PyCQA/pylint#4849]
### Bug description

Consider the following code snippet:
```python
"""p.py"""

class UnusedPrivateMember:
    """unused-private-member (W0238)"""

    @staticmethod
    def __private_method():
        """Is private and does nothing."""

    @classmethod
    def use_private_method(cls):
        """Calls private method."""
        cls.__private_method()
```

### Command used

```shell
pylint p.py
```

### Pylint output

```shell
W0238: Unused private member `UnusedPrivateMember.__private_method()` (unused-private-member)
```

### Expected behavior

Calling a private static method only on `cls` (reference to class) in a classmethod produces false positive: `unused-private-member`.

#4782 recently fixed some issues regarding the usage of private members on the class name. This one seems to have slipped through.

Thanks.

### Pylint version

```shell
pylint 2.10.0-dev0
astroid 2.6.6
Python 3.9.6 (default, Jun 30 2021, 10:22:16) 
[GCC 11.1.0]
```
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

**Comment by yumasheta** [comment]
I think the fix is easier than I thought. Though, please look at my PR, as I never used astroid before.
